File: leetcode/python/1291_sequential_digits.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> list[int]:
        MAX: int = 123456789
        res: list[int] = []
        for digits in range(len(str(low)), len(str(high)) + 1):
            unit = 10 ** (digits - 1)
            n = MAX // (10 ** (9-digits))
            for j in range(10 - digits):
                if n > high:
                    return res
                if n >= low:
                    res.append(n)
                n = (n%unit)*10 + (digits+1+j)
        return res

s = Solution()
print(s.sequentialDigits(10, 3000))
print(s.sequentialDigits(58, 123))

File: leetcode/rust/src/1291_sequential_digits.rs
use std::collections::VecDeque;

impl Solution {
    pub fn sequential_digits(low: i32, high: i32) -> Vec<i32> {
        let mut result = vec![];
        let mut queue = VecDeque::from([1, 2, 3, 4, 5, 6, 7, 8, 9]);
        while let Some(num) = queue.pop_front() {
            if num >= low && num <= high {
                result.push(num);
            }
            let last_digit = num % 10;
            if last_digit < 9 {
                queue.push_back(num * 10 + last_digit + 1);
            }
        }
        result
    }
}